cmd/lakectl/cmd/dbt_util_test.go

```diff
@@ -0,0 +1,263 @@
+			},
+			DbtResources: DBTResources,
+			ErrType:      nil,
+		},
+		{
+			Name: "command returns an error",
+			Args: args{
+				ProjectRoot:  "nevermind",
+				ResourceType: "nevermind",
+				SelectValues: []string{"nevermind"},
+				Executor:     DummyCommandExecutor{err: dummyCommandExecutorError{}, output: ""},
+			},
+			DbtResources: nil,
+			ErrType:      dummyCommandExecutorError{},
+		},
+		{
+			Name: "command returns invalid json",
+			Args: args{
+				ProjectRoot:  "nevermind",
+				ResourceType: "nevermind",
+				SelectValues: []string{"nevermind"},
+				Executor:     DummyCommandExecutor{err: nil, output: "invalid json"},
+			},
+			DbtResources: nil,
+			ErrType:      &json.SyntaxError{},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.Name, func(t *testing.T) {
+			dbtResources, err := DbtLsToJSON(tt.Args.ProjectRoot, tt.Args.ResourceType, tt.Args.SelectValues, tt.Args.Executor)
+			if (err != nil) && !errors.As(err, &tt.ErrType) {
+				t.Errorf("DbtLsToJSON() error = %v, expected ErrType: %v", err, tt.ErrType)
+				return
+			}
+			if !reflect.DeepEqual(dbtResources, tt.DbtResources) {
+				t.Errorf("DbtLsToJSON() dbtResources = %v, schemaName %v", dbtResources, tt.DbtResources)
+			}
+		})
+	}
+}
+
+func TestDbtRun(t *testing.T) {
+	schemaKey := "theschema"
+	schemaValue := "myschema"
+	type args struct {
+		ProjectRoot            string
+		Schema                 string
+		SchemaEnvVarIdentifier string
+		SelectValues           []string
+		Executor               CommandExecutor
+	}
+	type dbtRunTestCase struct {
+		Name      string
+		Args      args
+		RunOutput string
+		Err       error
+	}
+	tests := []dbtRunTestCase{
+		{
+			Name: "Sunny day flow",
+			Args: args{
+				ProjectRoot:            "nevermind",
+				Schema:                 schemaValue,
+				SchemaEnvVarIdentifier: schemaKey,
+				SelectValues:           nil,
+				Executor: DummyCommandExecutor{
+					err:       nil,
+					output:    "hello",
+					envReturn: true,
+				},
+			},
+			RunOutput: fmt.Sprintf("hello %s=%s", schemaKey, schemaValue),
+			Err:       nil,
+		},
+		{
+			Name: "schemaEnvVarIdentifier empty",
+			Args: args{
+				ProjectRoot:            "nevermind",
+				Schema:                 schemaValue,
+				SchemaEnvVarIdentifier: "",
+				SelectValues:           nil,
+				Executor: DummyCommandExecutor{
+					err:       nil,
+					output:    "hello",
+					envReturn: false,
+				},
+			},
+			RunOutput: "hello",
+			Err:       nil,
+		},
+		{
+			Name: "schema value empty",
+			Args: args{
+				ProjectRoot:            "nevermind",
+				Schema:                 "",
+				SchemaEnvVarIdentifier: schemaKey,
+				SelectValues:           nil,
+				Executor: DummyCommandExecutor{
+					err:       nil,
+					output:    "hello",
+					envReturn: false,
+				},
+			},
+			RunOutput: "hello",
+			Err:       nil,
+		},
+		{
+			Name: "command returns an error",
+			Args: args{
+				ProjectRoot:            "nevermind",
```

Comment:
when every arg is the same for all tests I usually remove it from the code and just pass it as argument